src/main/webapp/view/transcript/transcript.js

```diff
@@ -115,60 +86,109 @@ function seekTranscript(currentTime) {
+        transcriptLine.startTimestampMs, transcriptLine.endTimestampMs);
+
+    const template = document.getElementById(TRANSCRIPT_TEMPLATE);
+    this.attachShadow({mode: 'open'});
+    this.shadowRoot.appendChild(template.content.cloneNode(true));
+    this.updateTemplateSlot(
+        'timestamp-range', timestampRange, STYLE_TIME_RANGE);
+    this.updateTemplateSlot('content', transcriptLine.content, STYLE_CONTENT);
+
+    this.startTimestampMs = transcriptLine.startTimestampMs;
+    this.endTimestampMs = transcriptLine.endTimestampMs;
+    // Sets the current transcript line to be the first line.
+    if (currentTranscriptLine == null) {
+      currentTranscriptLine = this;
+    }
   }
-  transcriptLineLiElement.classList.add(BOLD_FONT_WEIGHT);
-  transcriptLineLiElement.classList.remove(DEFAULT_FONT_WEIGHT);
-}
 
-/**
- * Removes bold from the text in `transcriptLineLiElement` if it
- * is currently bolded.
- */
-function removeBold(transcriptLineLiElement) {
-  if (!isBolded(transcriptLineLiElement)) {
-    return;
+  /**
+   * Updates the template slot `slotName` with `slotValue` and styling
+   * from `slotStyle`.
+   */
+  updateTemplateSlot(slotName, slotValue, slotStyle) {
+    const span = document.createElement('span');
+    span.innerText = slotValue;
+    span.slot = slotName;
+    span.className = slotStyle;
+    this.appendChild(span);
   }
-  transcriptLineLiElement.classList.add(DEFAULT_FONT_WEIGHT);
-  transcriptLineLiElement.classList.remove(BOLD_FONT_WEIGHT);
-}
 
-/** Returns true if`transcriptLineLiElement` is bolded. */
-function isBolded(transcriptLineLiElement) {
-  return transcriptLineLiElement.classList.contains(BOLD_FONT_WEIGHT);
-}
+  /** Returns true if the text is already bolded. */
+  isBolded() {
+    return this.classList.contains(BOLD_FONT_WEIGHT);
+  }
 
-/**
- * Returns true if `currentTimeMs` is within the time range for
- * the current transcript line.
- */
-function isWithinCurrentTimeRange(currentTimeMs) {
-  return currentTranscriptLine.startTimestampMs <= currentTimeMs &&
-      currentTimeMs <= currentTranscriptLine.endTimestampMs;
-}
+  /**
+   * Bolds the text if it is not already
+   * bolded.
+   */
+  addBold() {
+    if (this.isBolded()) {
+      return;
+    }
+    this.classList.add(BOLD_FONT_WEIGHT);
+    this.classList.remove(DEFAULT_FONT_WEIGHT);
+  }
 
-/**
- * Scrolls `transcriptLine` to the top of the transcript area.
- * */
-function scrollToTopOfTranscript(transcriptLine) {
-  const transcriptContainer = document.getElementById(TRANSCRIPT_CONTAINER);
-  const ulElementOffset = transcriptLine.parentElement.offsetTop;
-  transcriptContainer.scrollTop = transcriptLine.offsetTop - ulElementOffset;
+  /**
+   * Removes bold from the text if it
+   * is currently bolded.
+   */
+  removeBold() {
+    if (!this.isBolded()) {
+      return;
+    }
+    this.classList.add(DEFAULT_FONT_WEIGHT);
+    this.classList.remove(BOLD_FONT_WEIGHT);
+  }
```

Comment:
Now that you're using classes, I feel like it would simplify things if `isBolded`, `addBold`, `removeBold`, were combined into a single `setBold(boolean)`.

Or even a `bold` setter and getter that adds/removes the styling and updates an internal state.

Reply:
Ah, sounds good!